### PR TITLE
config: deprecate --ticketmaxprice

### PR DESCRIPTION
**Pull request by dajohi**
See --ticketbuyer.maxpriceabsolute and --ticketbuyer.maxpricerelative.

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -891,6 +896,16 @@ func loadConfig() (*config, []string, error) {
 		maxPerBlock = cfg.TicketBuyFreq
 	}
 
+	// Use --ticketmaxprice setting if it has been set and --ticketbuyer.maxpriceabsolute
+	// has not been changed.
+	maxPriceAbsolute := cfg.TBOpts.MaxPriceAbsolute
+	if cfg.TicketMaxPrice.ToCoin() != defaultTicketMaxPrice &&
+		cfg.TBOpts.MaxPriceAbsolute.ToCoin() == defaultMaxPriceAbsolute.ToCoin() {
```
Comment (jrick) [inline config.go]: compare the embedded Amounts, not the floating-point representations of them

File: config.go [hunk 2]
```diff
@@ -891,6 +896,16 @@ func loadConfig() (*config, []string, error) {
 		maxPerBlock = cfg.TicketBuyFreq
 	}
 
+	// Use --ticketmaxprice setting if it has been set and --ticketbuyer.maxpriceabsolute
+	// has not been changed.
+	maxPriceAbsolute := cfg.TBOpts.MaxPriceAbsolute
```
Comment (jrick) [inline config.go]: This should be an Amount not an AmountFlag (which it is currently it shares the same type as MaxPriceAbsolute)

File: config.go [hunk 3]
```diff
@@ -66,6 +66,7 @@ const (
 	defaultExpiryDelta                      = 16
 	defaultFeeSource                        = ticketbuyer.TicketFeeMean
 	defaultAvgPriceMode                     = ticketbuyer.PriceTargetVWAP
+	defaultMaxPriceAbsolute  dcrutil.Amount = 0
```
Comment (jrick) [inline config.go]: This should be untyped so it can be compared to any type.

Comment (dajohi) [inline config.go]: done

File: config.go (hunk 2 above)
Comment (dajohi) [inline config.go]: done

**Comment by dajohi** [comment]
updated

**Pull request merged**